### PR TITLE
Add stress and pull/push tests [ISSUE-161]

### PR DESCRIPTION
**Pull request by ivanilves**
Solving https://github.com/ivanilves/lstags/issues/162

# GIF

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -26,11 +26,12 @@ before_script:
...
 
 script:
-  - make build
-  - make coverage
-  - if [[ "${TRAVIS_PULL_REQUEST}" == "false" ]]; then make blackbox-integration-test; fi
   - make lint
   - make vet
+  - make build
+  - make coverage
+  - make blackbox-integration-test
```
Comment (ivanilves) [inline .travis.yml]: Run it **always**!

File: Makefile [hunk 2]
```diff
@@ -57,6 +60,26 @@ shell-test-docker-tcp: DOCKER_HOST:=tcp://127.0.0.1:2375
 shell-test-docker-tcp:
 	./lstags nginx~/stable/
 
+shell-test-pullpush: start-local-registry push-to-local-registry stop-local-registry
+
+start-local-registry: REGISTRY_PORT:=5757
+start-local-registry:
+	docker rm -f registry-lstags &>/dev/null || true \
```
Comment (glensc) [inline Makefile]: `&>/dev/null` redirect is considered bashism.

Comment (ivanilves) [reply Makefile]: Yes, actually the whole line should be :fire:

File: .travis.yml [hunk 3]
```diff
@@ -26,11 +26,12 @@ before_script:
...
+  - make stress-test CONCURRENT_REQUESTS=96
```
Comment (vonrabbe) [inline .travis.yml]: I think we could do `CONCURRENT_REQUESTS=128` but OK...

File: Makefile [hunk 4]
```diff
@@ -1,12 +1,14 @@
...
-	lint vet fail-on-errors docker-image build xbuild changelog release validate-release deploy deploy-github deploy-docker \
-	poc-app wrapper install
+.PHONY: default PHONY clean offline prepare dep test unit-test whitebox-integration-test coverage blackbox-integration-test \
+	shell-test-alpine shell-test-wrong-image shell-test-docker-socket shell-test-docker-tcp shell-test-pullpush start-local-registry stop-local-registry push-to-local-registry \
+	stress-test lint vet fail-on-errors docker-image build xbuild changelog release validate-release deploy deploy-github deploy-docker poc-app wrapper install
 
 default: prepare dep test lint vet build
 
+PHONY:
+	@egrep "^[0-9a-zA-Z_\-]+:( |$$)" Makefile | cut -d":" -f1 | uniq | tr '\n' ' ' | sed 's/^/.PHONY: /;s/$$/\n/'
```
Comment (vonrabbe) [inline Makefile]: LOL

bash h4x0r!

**Pull request merged**